validate-infra/action.yml

```diff
@@ -0,0 +1,137 @@
+    required: false
+    default: ""
+  azure-subscription-id:
+    description: Azure service principal or managed identity subscription ID when using OIDC
+    required: false
+    default: ""
+  azure-tenant-id:
+    description: Azure service principal or managed identity tenant ID when using OIDC
+    required: false
+    default: ""
+  environment:
+    description: Environment to validate (test, platform-test, production)
+    required: true
+  terraform-base:
+    description: Path to the terraform files
+    required: false
+    default: "cluster/terraform_aks_cluster"
+  slack-webhook:
+    description: Slack webhook URL for notifications
+    required: false
+    default: ""
+
+outputs:
+  drift_detected:
+    description: Whether infrastructure drift was detected
+    value: ${{ steps.plan.outputs.drift_detected }}
+
+runs:
+  using: composite
+
+  steps:
+    - name: Checkout repository
+      uses: actions/checkout@v4
+      with:
+        fetch-depth: 0
+
+    - name: Set environment variables
+      shell: bash
+      run: |
+        # Get the current commit SHA (works for any ref)
+        GIT_COMMIT=$(git rev-parse HEAD)
+        echo "IMAGE_TAG=$GIT_COMMIT" >> $GITHUB_ENV
+        echo "DOCKER_IMAGE_TAG=$GIT_COMMIT" >> $GITHUB_ENV
+        echo "Using commit SHA: $GIT_COMMIT"
+
+    - name: Azure login
+      uses: azure/login@v2
+      with:
+        client-id: ${{ inputs.azure-client-id }}
+        tenant-id: ${{ inputs.azure-tenant-id }}
+        subscription-id: ${{ inputs.azure-subscription-id }}
+
+    - name: Set Environment variables
+      id: set_env_var
+      shell: bash
+      run: |
+        terraform_version=$(awk '/{/{f=/^terraform/;next}f' ${{ inputs.terraform-base }}/terraform.tf | grep -o [0-9\.]*)
+        echo "TERRAFORM_VERSION=$terraform_version" >> $GITHUB_ENV
+
+    - name: Use Terraform ${{ env.TERRAFORM_VERSION }}
+      uses: hashicorp/setup-terraform@v3
+      with:
+        terraform_version: ${{ env.TERRAFORM_VERSION }}
+        terraform_wrapper: false
+
+    - name: Set ARM environment variables
+      uses: DFE-Digital/github-actions/set-arm-environment-variables@master
+      with:
+        azure-client-id: ${{ inputs.azure-client-id }}
+        azure-subscription-id: ${{ inputs.azure-subscription-id }}
+        azure-tenant-id: ${{ inputs.azure-tenant-id }}
+
+    - name: Set kubelogin environment
+      uses: DFE-Digital/github-actions/set-kubelogin-environment@master
+      with:
+        azure-client-id: ${{ inputs.azure-client-id }}
+        azure-tenant-id: ${{ inputs.azure-tenant-id }}
+        azure-subscription-id: ${{ inputs.azure-subscription-id }}
+
+    - name: Run terraform plan with detailed exit code
+      id: plan
+      shell: bash
+      run: |
+        set +e  # Don't exit on non-zero return codes
+
+        echo "::group::Running terraform plan for ${{ inputs.environment }}"
+        DETAILED_EXITCODE=-detailed-exitcode make ci ${{ inputs.environment }} terraform-plan
```

Comment:
We should also test domains-plan and domains-infra-plan, so can we add extra steps for these.